cpp/inc/bond/core/apply.h

```diff
@@ -44,34 +46,48 @@ Apply(const Transform& transform, const T& value)
 /// @brief Apply transform which can modify an instance of a struct
 template <typename Transform, typename T>
 typename boost::enable_if<is_modifying_transform<Transform>, bool>::type inline
-Apply(const Transform& transform, T& value)
+ApplyTransform(const Transform& transform, T& value)
 {
     return StaticParser<T&>(value).Apply(transform, typename schema<T>::type());
 }
 
 
 // Specializations for transform requiring double-pass
 template <typename Transform, typename T, typename Reader>
-typename boost::enable_if<detail::need_double_pass<Transform>, bool>::type inline
-Apply(const Transform& transform, const bonded<T, Reader>& bonded)
+typename boost::enable_if<need_double_pass<Transform>, bool>::type inline
+ApplyTransform(const Transform& transform, const bonded<T, Reader>& bonded)
 {
     if (transform.NeedPass0())
-        return detail::DoublePassApply(transform, bonded);
+        return DoublePassApply(transform, bonded);
     else
         return bonded._Apply(transform);
 }
 
 
 template <typename Transform, typename T>
-typename boost::enable_if<detail::need_double_pass<Transform>, bool>::type inline
-Apply(const Transform& transform, const T& value)
+typename boost::enable_if<need_double_pass<Transform>, bool>::type inline
+ApplyTransform(const Transform& transform, const T& value)
 {
     if (transform.NeedPass0())
-        return detail::DoublePassApply(transform, value);
+        return DoublePassApply(transform, value);
     else
         return StaticParser<const T&>(value).Apply(transform, typename schema<T>::type());
 }
 
+} // namespace detail
+
+
+template <typename Transform, typename T>
+bool inline Apply(const Transform& transform, T& value)
```

Comment:
I think the only ones that won't be backward compatible are the overloads for `Marshaler<Writer>` where the first template argument is a `Writer` and not a `Transform`. Will put this in change log.